### PR TITLE
[FIx] Related state triple history

### PR DESCRIPTION
**Pull request by eilst**

### CODE REVIEW

File: project_event/models/project_task.py [hunk 1]
```diff
@@ -135,11 +135,33 @@ class Task(models.Model):
...
+        ('requested', 'Requested'),
+        ('read', 'Read'),
+        ('postponed', 'Postponed'),
+        ('accepted', 'Accepted'),
+        ('approved', 'Approved'),
+        ('done', 'Done'),
+        ('canceled', 'Canceled')],
+        string='State',
+        default='draft',
+        compute='_compute_task_state_report_done_required'
```
Comment (jananjoy) [inline project_event/models/project_task.py:150]: @eilst Not a good Method Name

File: project_event/models/project_task.py [hunk 2]
```diff
@@ -135,11 +135,33 @@ class Task(models.Model):
...
+        ('requested', 'Requested'),
+        ('read', 'Read'),
+        ('postponed', 'Postponed'),
+        ('accepted', 'Accepted'),
+        ('approved', 'Approved'),
+        ('done', 'Done'),
+        ('canceled', 'Canceled')],
+        string='State',
+        default='draft',
+        compute='_compute_task_state_report_not_done_required'
```
Comment (jananjoy) [inline project_event/models/project_task.py:164]: @eilst Not a good Method Name

File: project_event/models/project_task.py [hunk 3]
```diff
@@ -135,11 +135,33 @@ class Task(models.Model):
         default='draft',
         track_visibility='onchange',
     )
-    task_state_report_done_required = fields.Selection(
-        related='task_state'
+    task_state_report_done_required = fields.Selection([
```
Comment (jananjoy) [inline project_event/models/project_task.py:138]: @eilst  Two fields have the same structure except for the computing methods.  it seem like the code redundant

File: project_event/models/project_task.py [hunk 4]
```diff
@@ -407,6 +429,16 @@ def name_search(self, name='', args=None, operator='ilike', limit=100):
                       ('code', operator, name)]
         return super(Task, self).search(domain + args, limit=limit).name_get()
 
+    @api.depends('task_state')
+    def _compute_task_state_report_done_required(self):
```
Comment (jananjoy) [inline project_event/models/project_task.py:433]: @eilst Redundant function except the function signature

File: project_event/models/project_task.py [hunk 5]
```diff
@@ -407,6 +429,16 @@ def name_search(self, name='', args=None, operator='ilike', limit=100):
...
+        for rec in self:
+            rec.task_state_report_done_required = rec.task_state
+
+    @api.depends('task_state')
+    def _compute_task_state_report_not_done_required(self):
```
Comment (jananjoy) [inline project_event/models/project_task.py:438]: @eilst  Same comments as above

File: project_event/models/project_task.py (hunk 1 above)
Comment (eilst) [reply project_event/models/project_task.py:150]: what would you name it?

**Review by rimbendhaou — APPROVED** [review]

**Pull request merged**